tools/funccount.py

```diff
@@ -81,6 +85,16 @@ def signal_ignore(signal, frame):
 if debug:
     print(bpf_text)
 b = BPF(text=bpf_text)
+counts = b.get_table("counts")
+
+# pre-insert the function addresses into the counts table
+fns = b._get_kprobe_functions(pattern)
+for fn in fns:
+    addr = b.ksymname(fn)
+    if addr == -1:
+        raise Exception("Unknown symbol name %s" % fn)
+    counts[counts.Key(addr)] = counts.Leaf()
```

Comment:
Collision at which level? Two ksyms can have the same address (SyS_write and sys_write), but the address itself will have a unique entry in the bpf hlist underlying the map, correct? Are you suggesting to use BPF_NOEXIST? I don't think that would matter here.